modules/wfa-qt/src/indigo_api_callback_dut.c

```diff
@@ -1989,52 +1874,11 @@ static int configure_sta_handler(struct packet_wrapper *req, struct packet_wrapp
 
 static int associate_sta_handler(struct packet_wrapper *req, struct packet_wrapper *resp)
 {
-	struct wpa_ctrl *w = NULL;
-	struct wpa_supplicant *wpa_s = NULL;
-	size_t resp_len;
-	char buffer[32], response[16];
 	int status = TLV_VALUE_STATUS_OK;
 	char *message = TLV_VALUE_WPA_S_START_UP_OK;
 
-	wpa_s = z_wpas_get_handle_by_ifname(WIRELESS_INTERFACE_DEFAULT);
-	if (!wpa_s) {
-		indigo_logger(LOG_LEVEL_ERROR,
-			      "%s: Unable to get wpa_s handle for %s\n",
-			      __func__, WIRELESS_INTERFACE_DEFAULT);
-		status = TLV_VALUE_STATUS_NOT_OK;
-		message = TLV_VALUE_WPA_S_START_UP_NOT_OK;
-		goto done;
-	}
-
-	w = wpa_ctrl_open(wpa_s->ctrl_iface->sock_pair[0]);
-	if (!w) {
-		indigo_logger(LOG_LEVEL_ERROR, "Failed to connect to wpa_supplicant");
-		status = TLV_VALUE_STATUS_NOT_OK;
-		message = TLV_VALUE_WPA_S_START_UP_NOT_OK;
-		goto done;
-	}
-
-	memset(buffer, 0, sizeof(buffer));
-	sprintf(buffer, "ENABLE_NETWORK 0");
-	memset(response, 0, sizeof(response));
-	resp_len = sizeof(response) - 1;
-	wpa_ctrl_request(w, buffer, strlen(buffer), response, &resp_len, NULL);
-	if (strncmp(response, WPA_CTRL_OK, strlen(WPA_CTRL_OK)) != 0) {
-		indigo_logger(LOG_LEVEL_ERROR,
-			      "Failed to execute the command. Response: %s", response);
-		goto done;
-	}
-
-	memset(buffer, 0, sizeof(buffer));
-	sprintf(buffer, "SELECT_NETWORK 0");
-	memset(response, 0, sizeof(response));
-	resp_len = sizeof(response) - 1;
-	wpa_ctrl_request(w, buffer, strlen(buffer), response, &resp_len, NULL);
-	if (strncmp(response, WPA_CTRL_OK, strlen(WPA_CTRL_OK)) != 0) {
-		indigo_logger(LOG_LEVEL_ERROR,
-			      "Failed to execute the command. Response: %s", response);
-		goto done;
-	}
+	CHECK_RET(run_qt_command("ENABLE_NETWORK 0"));
```

Comment:
For error case this would return `TLV_VALUE_STATUS_OK`, please fix that.

Reply:
Being handled.